test/replicache_test.dart

```diff
@@ -657,4 +661,69 @@ Future<void> main() async {
     expect(deleteCount, 4);
     expect(createCount, 3);
   });
+
+  test('nested tx', () async {
+    await useReplay('nested tx');
+
+    rep = await replicacheForTesting('nested-tx');
+
+    expect(await rep.query((tx) async => 'abc'), 'abc');
+
+    expect(
+      rep.query((tx) async {
```

Comment:
I thought that we were going to allow this. It's only mutates that need to be protected.

Reply:
I guess I can see disallowing all nested transactions for simplicity.

Reply:
It catches programming errors. We do not want people to open a query and then read from a different Noms commit.